crates/storage/db/src/tables/mod.rs

```diff
@@ -245,7 +245,7 @@ table!(
 
 table!(
     /// Stores pointers to transactions that changed each storage key.
-    ( StorageHistory ) AddressStorageKey | TransitionList
+    ( StorageHistory ) StorageShardedKey | TransitionList
```

Comment:
wondering if this could be a `dupsort` table instead, so we don't repeat `Address` for every storage slot?

So instead of 
```
Address | ShardedKey<H256> -> TransitionList
```

it would become
```
Address -> ShardedKey<H256> | TransitionList
```

`ShardedKey<H256> | TransitionList` would have to be its own type though.  We could add a compress `ShardedKey<T>` impl that points to the `Encode` one.

Reply:
Probably yes, will introduce it. on every other storage (address,StorageKey) we used duptable.

Reply:
Where should i introduce the cut as we have:
`Address -> StorageKey -> BiggestTransitionId -> List`
Dup table can be:
`Address[MainKey] -> (StorageKey,BiggestTransitionId)[DupKey] -> List`
`(Address,StorageKey)[MainKey] -> BiggestTransitionId[DupKey] -> List`

Reply:
I'm leaning towards `(Address,StorageKey)[MainKey] -> BiggestTransitionId[DupKey] -> List` , but it's just a gut feeling and probably only applies to highly used contracts?